factory.py

```diff
@@ -13,7 +18,10 @@
 logger = logging.getLogger(__name__)
 
 
-def get_factories(task: Union[str, FactoryTypes, Tasks]) -> List["Factory"]:
+def get_factories(
+    task: Union[str, AdaptTasks, FactoryTypes],
+    job_service: Union[str, JobServices],
+) -> List["Factory"]:
```

Comment:
But shouldn't they be different PRs ? Why should they be included with this one ?

Reply:
such changes are expected in integration prs. and since we are adding adapt, I believe this is the correct time to ensure such changes.

Reply:
`class KompressTasks(StrEnum):
    TEXT_GENERATION = "llm"
    CLASSIFICATION = "image_classification"


class AdaptTasks(StrEnum):
    TEXT_GENERATION = "text_generation"
    TEXT_CLASSIFICATION = "text_classification"
    SEQ_2_SEQ_TRANSLATION = "Seq2Seq_tasks"
    SEQ_2_SEQ_SUMMARIZATION = "Seq2Seq_tasks"
    QUESTION_ANSWERING = "question_answering"
    IMAGE_CLASSIFICATION = "image_classification"
    OBJECT_DETECTION = "object_detection"
    IMAGE_SEGMENTATION = "image_segmentation"
    POSE_ESTIMATION = "pose_estimation"`



This is already present. I don't get what's the issue here

Reply:
by function signature I meant, it might be an overlook while implementing, but why is this method accepting a FactoryType as task argument. Lets have different tasks and use those instead of factory types if not needed

Reply:
The code is expecting FactoryTypes in the task variable in the main branch. That needs to change there I guess which will require a PR from your side. 
> Lets have different tasks and use those instead of factory types if not needed
This should have been thought of earlier.